src/components/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -235,16 +281,28 @@ export default class PeopleAndRoles extends Vue {
         // get ID of person to undo
         const id = person?.officer?.id
 
-        // get original person from original IA
-        const originalList = this.getOriginalIA.incorporationApplication?.parties || []
-        const originalPerson = originalList.find(x => x.officer.id === id)
+        // get value of original person from original IA
+        const originalPerson = this.originalList.find(x => x.officer.id === id)
 
         // safety check
         if (!originalPerson) {
-          console.log('Failed to find original person with id =', id) // eslint-disable-line no-console
+          // eslint-disable-next-line no-console
+          console.log('Failed to find original person with id =', id)
           return
         }
 
+        // check if original person had CP role and another person has it now
+        const hadCp = originalPerson.roles.some(role => role.roleType === Roles.COMPLETING_PARTY)
+        // get the Completing Party
+        // (we update this record right in the temp list)
+        const otherPerson = this.getCompletingParty(tempList)
+
+        if (hadCp && otherPerson && (originalPerson.officer.id !== otherPerson.officer.id)) {
+          // remove the other CP and their email
+          otherPerson.roles = otherPerson.roles.filter(r => r.roleType !== Roles.COMPLETING_PARTY)
+          otherPerson.officer.email = null
+        }
```

Comment:
This is the fix for the undo bug: the other person needs to no longer have role CP or email.

Reply:
If you're wondering what this functionality looks like: the other person is retained (along with their CORRECTED flag) and the original person is restored. I tested extensively and it feels OK.